cosmos/providers/dbt/core/operators.py

```diff
@@ -255,158 +225,22 @@ def build_and_run_cmd(self, env: dict, cmd_flags: list = None):
-        super().__init__(**kwargs)
-        self.base_cmd = "seed"
-
-    def add_cmd_flags(self):
-        flags = []
-        if self.full_refresh is True:
-            flags.append("--full-refresh")
-
-        return flags
-
-    def execute(self, context: Context):
-        cmd_flags = self.add_cmd_flags()
-        result = self.build_and_run_cmd(env=self.get_env(context), cmd_flags=cmd_flags)
-        return result.output
-
-
-class DbtRunOperator(DbtBaseOperator):
-    """
-    Executes a dbt core run command.
-
-    """
-
-    ui_color = "#7352BA"
-    ui_fgcolor = "#F4F2FC"
-
-    def __init__(self, **kwargs) -> None:
-        super().__init__(**kwargs)
-        self.base_cmd = "run"
-
-    def execute(self, context: Context):
-        result = self.build_and_run_cmd(env=self.get_env(context))
-        return result.output
-
-
-class DbtTestOperator(DbtBaseOperator):
-    """
-    Executes a dbt core test command.
-
-    """
-
-    ui_color = "#8194E0"
-
-    def __init__(self, **kwargs) -> None:
-        super().__init__(**kwargs)
-        self.base_cmd = "test"
-
-    def execute(self, context: Context):
-        result = self.build_and_run_cmd(env=self.get_env(context))
-        return result.output
-
-
-class DbtRunOperationOperator(DbtBaseOperator):
-    """
-    Executes a dbt core run-operation command.
-
-    :param macro_name: name of macro to execute
-    :type macro_name: str
-    :param args: Supply arguments to the macro. This dictionary will be mapped to the keyword arguments defined in the
-        selected macro.
-    :type args: dict
-    """
-
-    ui_color = "#8194E0"
-    template_fields: Sequence[str] = "args"
-
-    def __init__(self, macro_name: str, args: dict = None, **kwargs) -> None:
-        self.macro_name = macro_name
-        self.args = args
-        super().__init__(**kwargs)
-        self.base_cmd = ["run-operation", macro_name]
-
-    def add_cmd_flags(self):
-        flags = []
-        if self.args is not None:
-            flags.append("--args")
-            flags.append(yaml.dump(self.args))
-        return flags
-
-    def execute(self, context: Context):
-        cmd_flags = self.add_cmd_flags()
-        result = self.build_and_run_cmd(env=self.get_env(context), cmd_flags=cmd_flags)
-        return result.output
-
-
-class DbtDepsOperator(DbtBaseOperator):
-    """
-    Executes a dbt core deps command.
-
-    :param vars: Supply variables to the project. This argument overrides variables defined in your dbt_project.yml file
-    :type vars: dict
-    """
+        if handle_profile:
+            create_default_profiles(
+                DBT_PROFILE_PATH
+                if self.profiles_dir is None
+                else Path(self.profiles_dir).joinpath("profiles.yml")
+            )
+            profile, profile_vars = map_profile(
+                conn_id=self.conn_id,
+                db_override=self.db_name,
+                schema_override=self.schema,
+            )
 
```

Comment:
This should all be handled via the "DBT_PROFILES_DIR" environment variable. We have merged this since you've written this code so I do apologize for the inconvenience.

Reply:
Done